sentry_sdk/utils.py

```diff
@@ -773,32 +773,34 @@ def handle_in_app(event, in_app_exclude=None, in_app_include=None):
     return event
 
 
-def handle_in_app_impl(frames, in_app_exclude, in_app_include, default_in_app=True):
-    # type: (Any, Optional[List[str]], Optional[List[str]], bool) -> Optional[Any]
+def handle_in_app_impl(frames, in_app_exclude, in_app_include):
+    # type: (Any, Optional[List[str]], Optional[List[str]]) -> Optional[Any]
     if not frames:
         return None
 
-    any_in_app = False
     for frame in frames:
-        in_app = frame.get("in_app")
-        if in_app is not None:
-            if in_app:
-                any_in_app = True
+        current_in_app = frame.get("in_app")
+        if current_in_app is not None:
             continue
 
         module = frame.get("module")
-        if not module:
-            continue
-        elif _module_in_set(module, in_app_include):
+
+        if _module_in_list(module, in_app_include):
             frame["in_app"] = True
-            any_in_app = True
-        elif _module_in_set(module, in_app_exclude):
+
+        elif _module_in_list(module, in_app_exclude):
             frame["in_app"] = False
 
-    if default_in_app and not any_in_app:
-        for frame in frames:
-            if frame.get("in_app") is None:
-                frame["in_app"] = True
+        else:
+            abs_path = frame.get("abs_path")
+            if abs_path is None:
+                continue
+
+            external_source = (
+                re.search(r"[\\/](?:dist|site)-packages[\\/]", frame["abs_path"])
+                is not None
+            )
+            frame["in_app"] = not external_source
```

Comment:
This sets anything that's not in site packages or dist packages to false. Including frames coming from the standard library which is unexpected at least from the profiling side.

Reply:
Valid point. Fixed.